Documentation/gettingstarted/k8s-install-rke.rst

```diff
@@ -0,0 +1,131 @@
+  - `Helm`_ Version 3
+
+Please consult `RKE Requirements`_ and Cilium :ref:`admin_system_reqs`.
+
+.. _RKE Requirements: https://rancher.com/docs/rke/latest/en/os/
+.. _Helm: https://helm.sh
+
+Installing Rancher Kubernetes Engine
+====================================
+
+Install the RKE Binary according to the `RKE Installation Guide <https://rancher.com/docs/rke/latest/en/installation/>`_
+
+Alternatively, on macOS the binary can be installed using Homebrew:
+
+.. code:: bash
+
+  $ brew install rke
+
+If you have a previous version of RKE installed, you can update it:
+
+
+.. code:: bash
+
+  $ brew upgrade rke
+
+Prepare the Kubernetes Cluster
+==============================
+
+Prepare your hosts with a supported operating system and Docker runtime. This
+is outside the scope of this guide.
+
+Prepare a RKE Configuration
+---------------------------
+
+Once your nodes are running and have been configured with the docker runtime,
+the next step is to generate a `RKE Configuration File <https://rancher.com/docs/rke/latest/en/installation/#creating-the-cluster-configuration-file>`_
+
+You can create a new cluster.yaml file using prompts:
+
+.. code:: bash
+
+  $ rke config --name cluster.yaml
+
+You can also use the `minimal example <https://rancher.com/docs/rke/latest/en/example-yamls/#minimal-cluster-yaml-example>`_
+
+When specifying the Network Provider Type enter none.
+
+.. code-block:: yaml
+
+  {network: {plugin: none}}
+
+
+Launch the RKE Cluster
+----------------------
+
+Once the configuration file is in place, launch the RKE Kubernetes cluster:
+
+.. code:: bash
+
+  $ rke up
+
+If your config uses a filename different than cluster.yaml:
+
+.. code:: bash
+
+  $ rke up --config <filename>
+
+Once the cluster is launched, the installer will create a file in the local
+directory named kube_config_<FILE_NAME>.yml which can be used to access the
+Kubernetes API server.
+
+.. code:: bash
+
+  $ export KUBECONFIG=$PWD/kube_config_cluster.yml
+  $ kubectl get nodes
+
+Installing Cilium
+=================
+
+Cilium installation requires `Helm`_ version 3. You can install Helm according
+to the `Helm Installation Guide <https://docs.helm.sh/docs/intro/install/>`_.
+
+Next, add the Cilium helm repository:
+
+.. code:: bash
+
+  $ helm repo add cilium https://helm.cilium.io
+
+Finally, install Cilium as your CNI:
+
+.. tabs::
+  .. group-tab:: Helm
+
+    .. code:: bash
+
+     $ helm install --namespace kube-system cilium cilium/cilium
+
+  .. group-tab:: kubectl
+
+    .. code:: bash
+
+      $ helm template --namespace kube-system cilium/cilium > cilium.yaml
+      $ kubectl apply -f cilium.yaml
+
+.. include:: k8s-install-validate.rst
+.. include:: hubble-install.rst
+.. include:: getting-started-next-steps.rst
+
+Now that you have a Kubernetes cluster with Cilium up and running, you can take
+a couple of next steps to explore various capabilities:
+
+* :ref:`gs_http`
+* :ref:`gs_dns`
+* :ref:`gs_cassandra`
+* :ref:`gs_kafka`
```

Comment:
My vagrant setup just seem to kind of bog down after cilium daemonset was installed.  I was able to run kubectl get pods once, but then not after that.  Cilium had initially been in an error state, and came up, but the other pods were still stuck in "container creating" 

```
cattle-system   cattle-cluster-agent-77b98bc75b-ps2pp     0/1     ContainerCreating   0          7m17s
cattle-system   cattle-node-agent-45hjs                   1/1     Running             0          7m16s
cattle-system   kube-api-auth-nktdb                       1/1     Running             0          7m14s
ingress-nginx   default-http-backend-67cf578fc4-jslf4     0/1     ContainerCreating   0          7m49s
ingress-nginx   nginx-ingress-controller-d5lxw            1/1     Running             0          7m49s
kube-system     cilium-8fjct                              1/1     Running             1          2m48s
kube-system     cilium-operator-59b8b544d4-79b8c          1/1     Running             0          2m48s
kube-system     coredns-7c5566588d-wpq9l                  0/1     ContainerCreating   0          8m2s
kube-system     coredns-autoscaler-65bfc8d47d-zr2x8       0/1     ContainerCreating   0          8m
kube-system     metrics-server-6b55c64f86-j4c86           0/1     ContainerCreating   0          7m52s
kube-system     rke-coredns-addon-deploy-job-n2xg7        0/1     Completed           0          8m6s
kube-system     rke-ingress-controller-deploy-job-vq58g   0/1     Completed           0          7m50s
kube-system     rke-metrics-addon-deploy-job-srzpn        0/1     Completed           0          7m56s
```

Further attempts to use kubectl took minutes.  Seemed like maybe there was a webhook or something similar that was failing.

Reply:
Update:   Yes, a lot of the pods were failing to deploy with the following error: 

  Warning  NetworkNotReady  9m10s (x151 over 14m)  kubelet, node-01   network is not ready: runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized

This would be expected before Cilium was installed, but kind of odd for after the fact.  Does kubelet need to be restarted to pick up a new CNI file?  If so, we may need: 

```
  --set global.nodeinit.enabled=true \
  --set nodeinit.reconfigureKubelet=true \
```

passed to helm

Reply:
Also, seems like BPF filesystem is not mounted (at least when running without node-init).  I think there are also mechanisms in nodeinit to handle that: 

```
level=warning msg="BPF filesystem is going to be mounted automatically in /run/cilium/bpffs. However, it probably means that Cilium is running inside container and BPFFS is not mounted on the host. for more information, see: https://cilium.link/err-bpf-mount" subsys=bpf
level=warning msg="================================= WARNING ==========================================" subsys=bpf
level=warning msg="BPF filesystem is not mounted. This will lead to network disruption when Cilium pods" subsys=bpf
level=warning msg="are restarted. Ensure that the BPF filesystem is mounted in the host." subsys=bpf
level=warning msg="https://docs.cilium.io/en/stable/kubernetes/requirements/#mounted-bpf-filesystem" subsys=bpf
level=warning msg="====================================================================================" subsys=bpf
```

Reply:
^^^ is obviously specific to whatever OS they are running in their vagrant setup, and might not apply to other based OSes used with rancher.

Reply:
See above related to OS and potentially locally impacting problems. Since this is agnostic to any OS and uses SSH as an interface, it should work for most people who are using any worker host that can be accessed via SSH and is supported by RKE/Cilium.